modules/onerepo/src/core/install/commands/install.ts

```diff
@@ -42,91 +46,129 @@ export const builder: Builder<Args> = (yargs) =>
-				argv.location = os.platform() === 'darwin' ? '/usr/local/bin' : `${homedir}/bin`;
+				argv.location = `${homedir}/.onerepo`;
 			}
-		})
-		.epilogue(
-			'`npx something-something`? `npm run what`? `yarn that-thing`? `../../../bin/one`? Forget all of that; no more will you need to figure out how to run your CLI. Just install it directly into your user bin PATH with this command.',
-		)
-		.epilogue(
-			'As an added bonus, tab-completions will be added to your .zshrc or .bash_profile (depending on your current shell).',
-		);
+		}, true);
 
-export const handler: Handler<Args> = async function handler(argv, { graph, logger }) {
-	const { force, location, name } = argv;
+export const handler: Handler<Args> = async function handler(argv, { logger }) {
+	const { force, location, version } = argv;
 
 	if (!force) {
-		const step = logger.createStep(`Ensure ${name} does not exist`);
-		const [which] = await run({
-			name: `which ${name}?`,
-			cmd: 'which',
-			args: [name],
-			step,
-			skipFailures: true,
-		});
-
-		if (which) {
-			const contents = await file.read(which, 'r', { step });
-			if (!contents.includes(process.argv[1])) {
-				step.error(`Refusing to install with name \`${name}\` because it already exists at \`${which}\`
-
-				To force installation, re-run this command with \`--force\`:
-					${path.relative(process.cwd(), process.argv[1])} ${command} --name=${name} --location=${location} --force`);
+		const step = logger.createStep('Check version');
+		if (await file.exists(path.join(location!, 'package.json'), { step })) {
+			const curPkgJsonRaw = await file.read(path.join(location!, 'package.json'), constants.O_RDONLY, {
+				step,
+			});
+			const pkgJson = JSON.parse(curPkgJsonRaw);
+			const comp = compare(pkgJson.version, version);
+			if (comp > 0) {
+				step.error(`Current version of oneRepo (${pkgJson.version}) is greater than the requested version (${version}).
+Override this check by re-running with "--force".`);
 				await step.end();
 				return;
 			}
-
-			step.warn(
-				`"${name}" has already been installed. It should be safe to overwrite previous installation. Continuing…`,
-			);
 		}
-		await step.end();
+		step.end();
 	}
 
-	const installLocation = path.join(location!, name);
+	const installStep = logger.createStep('Install');
 
-	await sudo({
-		name: 'Create executable',
-		cmd: 'echo',
-		args: [`"#!/bin/zsh\n\n${process.argv[1].replace(/ /g, '\\ ')} \\$@"`, '|', 'sudo', 'tee', installLocation],
+	await file.write(
+		path.join(location!, 'package.json'),
+		JSON.stringify(
+			{
+				name: 'onerepo-bin',
+				version: version,
+				type: 'module',
+				workspaces: [],
+				dependencies: {
+					onerepo: version,
+				},
+			},
+			null,
+			2,
+		),
+		{ step: installStep },
+	);
+
+	await file.write(
+		path.join(location!, 'onerepo.config.js'),
+		`
+export default {
+	root: true,
+};
+`,
+		{ step: installStep },
+	);
+
+	await run({
+		name: 'Installing dependencies',
+		cmd: 'npm',
```

Comment:
no yarn for installing here?

Reply:
Yeah I specifically am doing `npm` for now, because it's still guaranteed to be available if `node` is installed – (unless you've built node from source and specifically not added npm, which is out of scope for us to care about here).

Keep in mind, this install is _outside_ of any repository the user manages.

We _could_ use the `@onerepo/package-manager`, which looks at `npm_config_user_agent` to determine what you're currently using, but that may also require running `corepack`, so there may be some indirection that I'd rather not deal with.